WootricSDK/WootricSDK/WTRApiClient.m

```diff
@@ -246,14 +238,35 @@ - (void)createResponseForEndUser:(NSInteger)endUserID withScore:(NSInteger)score
   [dataTask resume];
 }
 
+- (void)getRegisteredEventList:(void (^)(NSArray *))events {
+  NSString *baseURLString = [NSString stringWithFormat:@"%@%@?account_token=%@",
+                             WTRSurveyServerURL, WTRRegisterEventsEndpoint, _accountToken];
+
+  NSURL *url = [NSURL URLWithString:baseURLString];
+  NSMutableURLRequest *urlRequest = [NSMutableURLRequest requestWithURL:url];
+  [urlRequest setValue:@"Wootric-Mobile-SDK" forHTTPHeaderField:@"User-Agent"];
+
+  [WTRLogger log:@"eligibility - %@", urlRequest];
+
+  NSURLSessionDataTask *dataTask = [_wootricSession dataTaskWithRequest:urlRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+    if (error) {
+      [WTRLogger logError:@"%@", error];
+    } else {
+      NSArray *responseJSON = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+      if (responseJSON) {
+        events(responseJSON);
```

Comment:
It would be a good idea to store the list of events here so you are not calling eligibility at all times. Try to keep them at least for the session of the app or until it's minimized